projects/pastanaga-angular/src/lib/controls/textfield/select/select.component.ts

```diff
@@ -0,0 +1,373 @@
+        return (
+            !!changes.required ||
+            !!changes.maxlength ||
+            !!changes.pattern ||
+            !!changes.errorMessage
+        );
+    };
+
+    refreshInternalValidators() {
+        this.requiredValidator = this.required ? Validators.required : undefined;
+        this.patternValidator = !!this.pattern ? Validators.pattern(this.pattern) : undefined;
+        this.maxlengthValidator = !!this.maxlength ? Validators.maxLength(this.maxlength) : undefined;
+    }
+
+    filterOptions(optionLabel?: string | number) {
+        if (this.control.pristine) {
+            return;
+        }
+        const displayedValue = `${optionLabel}`.toLocaleLowerCase();
+        this.hideIrrelevantOptions(displayedValue);
+    }
+
+    hideIrrelevantOptions(displayedValue: any) {
+        if (!!this.ngContentOptions && this.ngContentOptions.length > 0) {
+            this.ngContentOptions.forEach((option) => {
+                option._hidden = !option.text.toLocaleLowerCase().includes(displayedValue);
+                option.refresh();
+            });
+            markForCheck(this.cdr);
+        }
+    }
+
+    selectInputChanged(optionLabel?: string | number) {
+        this.control.markAsDirty();
+        this.filterOptions(optionLabel);
+    }
+
+    toggleDropdown() {
+        if (this.control.disabled || this._readonly) {
+            return;
+        }
+        if (!!this.optionsDropdown && this.optionsDropdown._isDisplayed) {
+            this.optionsDropdown.close();
+        } else {
+            this.clickOnSelectInput();
+        }
+    }
+
+    /**
+     * click on select input should trigger paPopup toggle and display the options
+     * paPopup will listen to click and close itself immediately if we dont wait for a tick
+     */
+    clickOnSelectInput() {
+        setTimeout(() => {
+            if (!!this.selectInput) {
+                this.selectInput.nativeElement.click();
+            }
+        }, 0);
+    }
+
+    selectListOption(selectedOption: OptionModel) {
+        if (!selectedOption.disabled && this.isActive()) {
+            this.onValueChange(selectedOption.value);
+        }
+    }
+
+    onBlur() {
+        this.applyUserInput();
+    }
+
+    onEnter() {
+        if (!!this.optionsDropdown && this.optionsDropdown._isDisplayed) {
+            this.optionsDropdown.close();
+        }
+        if (!!this.selectInput) {
+            this.selectInput.nativeElement.blur();
+        }
+    }
+
+    applyUserInput() {
+        if (!this.isActive()) {
+            return;
+        }
+        const optionValueForUserInput = this.findValueByLabel();
+        if (!!optionValueForUserInput) {
+            this.onValueChange(optionValueForUserInput);
+        } else if (this.suggestionMode) {
+            const sanitized = sanitizeStringValue(this.displayedValue, this._acceptHtmlTags);
+            this.onValueChange(sanitized);
+        } else {
+            this.onValueChange(null);
+        }
```

Comment:
ok to keep this like that for now, but I still feel like we should ask confirmation to UX about that behaviour: if I understand well, there is no way to just filter the options displayed on the dropdown so user can select one.

Reply:
options are filtered when typing. On blur though... if no option matches what the user typed, the field is emptied and the value is set to null...
it sounds better than keeping something printed in the input while the selected option is different.
I agree though that we could also repaint the selected label if any and yeas, that's a UX question...

Reply:
right, ok I didn't understood it that way. So the behavior you have is good, except if an option was already selected and what user typed didn't select any option: in that case you should not remove the existing selection, you should remove what the user typed and replace it with existing selection. (No need to ask UX on that one, that will be raised as a bug if we let it like this)